### PR TITLE
Examples: Introduce `<skip>` for our examples + Add some Stackblitz examples + undupe code + homogenize our design for the examples in the doc

### LINKED ISSUE #36391: [Tracking] Fix StackBlitz examples
Some of the examples doesn't work when they are edited in StackBlitz due to specific CSS or JS.

This issue replaces https://github.com/twbs/bootstrap/pull/36091 and contains:
- the list of broken examples
- the list of tasks planned to fix those examples
- the references in PRs and issues already created

### Step-by-step fix

- [ ] Embed `snippets.js` (full or parts of it) in StackBlitz environment to fix examples that need specific JS. Tackled by https://github.com/twbs/bootstrap/pull/36127 or alternative https://github.com/twbs/bootstrap/pull/36352
- [ ] Embed specific CSS in StackBlitz environment to fix examples that need specific CSS.
- [ ] Hide some Popover and Tooltip examples (see https://github.com/twbs/bootstrap/pull/36127#discussion_r851609188) discussion. We could use a specific param as tried in https://github.com/twbs/bootstrap/blob/89cdd8a6ec1a00c24bbbd4dda1a08335095a78ca/site/layouts/shortcodes/example.html.

#### PRs
- https://github.com/twbs/bootstrap/pull/36091
- https://github.com/twbs/bootstrap/pull/36127 or https://github.com/twbs/bootstrap/pull/36352

#### Issues
- https://github.com/twbs/bootstrap/issues/36364

### List of examples

- [x] Accordions (3 examples) 
- [x] Alerts (7 examples)
  - [ ] [Live example](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/alerts/#live-example): hidden for the moment to test the hidden feature. TODO: try to embed the JS.
- [x] Badge - 6 examples
- [x] Breadcrumb - 4 examples
- [x] Button Group - TODO
  - [Sizing](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/button-group/#sizing) and [Vertical variation](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/button-group/#vertical-variation) could be examples 
- [x] Buttons - 15 examples
- [x] Card - 30 examples
  - In all card examples, the image text is not centered nor with the good size 
- [x] Carousel - 8 examples
- [x] Close Button - 3 examples
  - [White variant](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/close-button/#white-variant): We could pass a parameter to define a background color (here black). 
- [x] Collapse - 3 examples
- [x] Dropdowns - 21 examples
  -  Colored dropdowns don't have their edit button
  - [Split button](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#split-button) doesn't have any edit button
  - [Sizing](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#sizing) doesn't have any edit button
  - [Dropup](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#dropup) doesn't have any edit button
  - [Dropend](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#dropend) doesn't have any edit button
  - [Dropstart](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#dropstart) doesn't have any edit button
  - [ ] [Menu items 2](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#menu-items) is not working because of the `display: none` of `.dropdown-menu`.
  - [ ] [Menu items > Active](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#active)  is not working because of the `display: none` of `.dropdown-menu`.
  - [ ] [Menu items > Disabled](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#disabled)  is not working because of the `display: none` of `.dropdown-menu`.
  - [ ] [Menu Content - Headers](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#headers)  is not working because of the `display: none` of `.dropdown-menu`.
  - [ ] [Menu Content - Dividers](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#dividers)  is not working because of the `display: none` of `.dropdown-menu`.
  - [ ] [Menu Content - Text](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#text) is not working because of the `display: none` of `.dropdown-menu`.
  - [ ] [Menu Content - Forms](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/dropdowns/#forms) both examples are not working because of the `display: none` of `.dropdown-menu`.
- [x] List Group - 15 examples
- [x] Modal - 2 examples
  - Maybe some other examples could be edited
- [x] Navbar - 26 examples
  - [ ] [Image](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/navbar/#image) and [Image and text](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/navbar/#image-and-text) are not working because of the image relative path. Could be modified in the shortcode?
  - [Toggler](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/navbar/#toggler) Don't see the toggler but I suppose this is because of a difference between 5.1 and 5.2
- [x] Nav Tabs - 15 examples
- [x] Offcanvas - 8 examples
  - [ ] [Offcanvas components](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/offcanvas/#offcanvas-components) is not working
  - Check https://github.com/twbs/bootstrap/pull/36151
- [x] Pagination - 8 examples
- [x] Placeholders - 5 examples
- [x] Popovers - 5 examples
  - [ ] None of the examples are working - see https://github.com/twbs/bootstrap/pull/36127
- [x] Progress - 8 examples
- [x] Spinners - 13 examples
- [x] Toasts - 10 examples
  - Funny but good to know. If you click on the cross in the doc and then click on the "Try it" button, the environment won't display the toast.
  - [ ] [Placement](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/components/toasts/#placement): changing toast placement doesn't move the toast. The second example is not working as well
- [x] Tooltips - 2 examples
  - [ ] None of the examples are working
- [x] Figures - 2 examples
  - [ ] "400x300" are not centered correctly as in the cards and carousels 
- [x] Images - 5 examples
  - [ ] Same observation as for the Figures 
- [x] Reboot - 8 examples
- [x] Tables - 2 examples
- [x] Typography - 12 examples
- [x] Forms > Check Radios - 17 examples
  - [ ] [Indetermediate](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/forms/checks-radios/#indeterminate) doesn't seem to work 
- [x] Forms > Floating Labels - 8 examples
- [x] Forms > Form Control - 9 examples
- [x] Forms > Input Group - 11 examples
- [x] Forms > Layout - 10 examples
- [x] Forms > Overview - 4 examples
- [x] Forms > Range - 4 examples
- [x] Forms > Select - 5 examples
- [x] Forms > Validation - 6 examples
  - [ ] [Custom validation](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/forms/validation/#custom-styles) doesn't work because of the missing JS explained just after 
  - [ ] [Tooltips](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/forms/validation/#tooltips) doesn't work (see Tooltips section probably)
- [x] Helpers > Clearfix - 1 example
- [x] Helpers > Colored links - 1 example
- [x] Helpers > Ratio - 4 examples
  - [ ] [Aspect ratios](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/helpers/ratio/#aspect-ratios) and [Custom ratios](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/helpers/ratio/#custom-ratios) are not working because of the specific CSS for `.bd-example-ratios .ratio`
- [x] Helpers > Stacks - 6 examples
- [x] Helpers > Stretched Link - 4 examples
- [x] Helpers > Text Truncation - 1 example
- [x] Helpers > Vertical rule - 3 examples
- [x] Helpers > Visually hidden - 1 example
- [x] Layout > Columns - 13 examples
  - [ ] Issues with the rendering because of `.bd-example-row-flex-cols .row` rules 
- [x] Layout > CSS Grid - 15 examples
  - [ ] Issues with the rendering because of `.bd-example-cssgrid .grid > *`
- [x] Layout > Grid - 15 examples
  - [ ] Issues with the rendering because of `.bd-example-row .row > .col, .bd-example-row .row > [class^="col-"]` 
- [x] Layout > Gutters - 6 examples
  - [ ] Some issues with the rendering of some examples because of `.bd-example-row .row > .col, .bd-example-row .row > [class^="col-"]` 
- [x] Utilities > Background - 3 examples
- [x] Utilities > Borders - 8 examples
  - [ ] Issues with rendering because of `.bd-example-border-utils [class^="border"]` specific rule 
- [x] Utilities > Colors - 3 examples
- [x] Utilities > Display - 4 examples
- [x] Utilities > Flex - 12 examples
  - [ ] Some issues with the rendering because of `.bd-example` specific rule
- [x] Utilities > Float - 2 examples
- [x] Utilities > Interactions - 2 examples
- [x] Utilities > Position - 5 examples
  - [ ] Issues with rendering because of `.bd-example-position-utils .position-absolute` specific rule 
- [x] Utilities > Shadow - 1 example
- [x] Utilities > Sizing - 4 examples
- [x] Utilities > Spacing - 1 example
- [x] Utilities > Text - 11 examples
  - [ ] [Text wrapping and overflow](https://deploy-preview-36091--twbs-bootstrap.netlify.app/docs/5.1/utilities/text/#text-wrapping-and-overflow): the 2nd example doesn't work because of `.bd-highlight` coming from the docs CSS 
- [x] Utilities > Vertical Align - 2 examples

### PR DESCRIPTION
**Pull request by louismaximepiton**
### Description

⚠️ This PR can be split into many if needed ⚠️ 
First part is #39328. Rebase once merged.

Rewrite examples shortcodes. -> #39328.
Homogenize the examples over the doc. -> #39328.
Add some missing use cases in docsref page. -> #39328.
Change the CSS for example. -> #39328.
Some proposals to better the doc, meaning having some more Stackblitz examples.
Avoid having some dupe code to maintain
Introduce the use of `<skip>` for example shortcode (not visible from anywhere for end-user). -> #39747.

<details>
  <summary>More details about the files changed</summary>

#### Changes that happen in the doc
- [Homepage](https://deploy-preview-38933--twbs-bootstrap.netlify.app/) -> highlight next to Read the doc button
- Code example inside callouts
- [Content > Typography #Headings](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/content/typography#headings) and [display headings](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/content/typography#display-headings)
- [Customize > Color modes #Custom-color-modes](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/customize/color-modes#custom-color-modes)
- [Customize > Components #creating-your-own](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/customize/components#creating-your-own)
- [Docsref page](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/docsref)

#### New working StackBlitz example
- [Forms validation](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/forms/validation) -> 2 new examples
- [Buttons](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/buttons#variables) -> 1 new example
- [Dropdowns](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/dropdowns) -> 7 new examples
- [List group](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/list-group) -> 1 new example
- [Modal](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/modal) -> 10 new examples
- [Navbar](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/navbar) -> 3 new examples
- [Navs&Tabs](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/navs-tabs) -> 4 new examples
- [Placeholders](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/placeholders) -> 1 new example
- [Scrollspy](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/scrollspy) -> 4 new examples
- [Toasts](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/toasts) -> 1 new example
- [Tooltips](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/tooltips) -> 2 new examples
- [Flex utility](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/utilities/flex) -> 12 new examples
- [Opacity utility](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/utilities/opacity) -> 1 new example
- [Overflow utility](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/utilities/overflow) -> 3 new examples
- [Spacing utility](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/utilities/spacing) -> 1 new example
- [Reboot](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/content/reboot) -> 7 new examples
- [Table]https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/content/tables) -> 37 new examples

#### Still some issues on them
- [Carousel](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/carousel) aren't initialized on StackBlitz but still work fine once initialized.
- [Scrollspy](https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/scrollspy) aren't initialized on StackBlitz but still work fine once initialized.
</details>

### Motivation & Context

Homogenize handling of our examples
Propose some missing examples to Stackblitz

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

#### Live previews

Some examples of implementation
- <https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/buttons#variables>
- <https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/dropdowns#sizing>
- <https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/modal#live-demo>
- <https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/components/navs-tabs#javascript-behavior>

Check that the rendering of this page look good.
- <https://deploy-preview-38933--twbs-bootstrap.netlify.app/docs/5.3/docsref>

### Related issues

Might close #36391.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Can you please fix your PR title and commits to be more descriptive please?

File: site/content/docs/5.3/components/list-group.md [hunk 1]
```diff
@@ -335,55 +335,34 @@ Loop that generates the modifier classes with an overriding of CSS variables.
...
+        <p>Some placeholder content in a paragraph relating to "Messages". And some more content, used here just to pad out and fill this tab panel. In production, you would obviously have more real content here. And not just text. It could be anything, really. Text, images, forms.</p>
+      </skip>
+      <skip class="tab-pane fade" id="list-settings" role="tabpanel" aria-labelledby="list-settings-list">
+        <p>Some placeholder content in a paragraph relating to "Settings". And some more content, used here just to pad out and fill this tab panel. In production, you would obviously have more real content here. And not just text. It could be anything, really. Text, images, forms.</p>
+      </skip>
     </div>
   </div>
 </div>
-```
+  {{< /example >}}
```
Comment (XhmikosR) [inline site/content/docs/5.3/components/list-group.md:365]: indentation is off, there might be more places

File: site/content/docs/5.3/components/navbar.md [hunk 2]
```diff
@@ -109,7 +109,7 @@ You can replace the text within the `.navbar-brand` with an `<img>`.
 <nav class="navbar bg-body-tertiary">
   <div class="container">
     <a class="navbar-brand" href="#">
-      <img src="/docs/{{< param docs_version >}}/assets/brand/bootstrap-logo.svg" alt="Bootstrap" width="30" height="24">
+      <img src="https://getbootstrap.com/docs/{{< param docs_version >}}/assets/brand/bootstrap-logo.svg" alt="Bootstrap" width="30" height="24">
```
Comment (XhmikosR) [inline site/content/docs/5.3/components/navbar.md:112]: This kind of changes should be reverted. We need the files locally referenced.

**Comment by XhmikosR** [comment]
Please split this kind of PRs into smaller ones. You have all kinds of different changes. I also left some comments from a very quick review.

**Comment by louismaximepiton** [comment]
Most of the PR has been split.

**Pull request closed without merging**